pade/core/agent.py

```diff
@@ -810,18 +806,45 @@ def react(self, message):
             if not self.ignore_ams or not message.system_message:
                 self.messages.appendleft(message)
 
+
+    def receive(self, message_filter = None):
+        '''
+        Get the first message from the top of the message stack (left-side of
+        the deque) with match the filters provided.
+
+        If any filter was provided, return the message from the top of stack.
+        
+        Parameters
+        ----------
+        message_filter : Filter
+            filter to be applied at the messages
+        '''
+
+        with self.messages_lock:
+            if message_filter == None:
+                return self.messages.popleft()
+
+            for message in self.messages:
+                if message_filter.filter(message):
+                    self.messages.remove(message)
+                    return message
+        return None
+
+
     def setup(self):
         ''' This method is an alternative to initiate agents without
         override the self.on_start() method in the subclasses.
         '''
         pass
 
+
```

Comment:
Just a small thing, to put it PEP8 compatible, [uses only one blank line](https://www.python.org/dev/peps/pep-0008/#blank-lines) to methods inside the class.

There are other examples below and above this comment.

Reply:
Resolved.